opm/parser/eclipse/Parser/Parser.hpp

```diff
@@ -101,6 +102,29 @@ namespace Opm {
             addParserKeyword( std::unique_ptr< ParserKeyword >( new T ) );
         }
 
+        // below are some helper functions
+
+        static EclipseState parse(std::shared_ptr<const Deck> deck, const ParseContext& context = ParseContext());
```

Comment:
Unsure about this - with the `Deck` the input is already parsed, and his is *very* similar to the `EclipseState` constructor. Does it have to exist?